components/brave_ads/core/browser/service/virtual_pref_provider.cc

```diff
@@ -0,0 +1,186 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/browser/service/virtual_pref_provider.h"
+
+#include <cstddef>
+#include <utility>
+#include <vector>
+
+#include "base/check.h"
+#include "base/json/json_reader.h"
+#include "base/version.h"
+#include "base/version_info/version_info.h"
+#include "brave/components/brave_ads/core/public/common/locale/locale_util.h"
+#include "brave/components/skus/browser/pref_names.h"
+#include "components/prefs/pref_service.h"
+
+namespace brave_ads {
+
+namespace {
+
+constexpr char kSkuEnvironmentPrefix[] = "skus:";
+constexpr char kSkuOrdersKey[] = "orders";
+constexpr char kSkuOrderLocationKey[] = "location";
+constexpr char kSkuOrderCreatedAtKey[] = "created_at";
+constexpr char kSkuOrderExpiresAtKey[] = "expires_at";
+constexpr char kSkuOrderLastPaidAtKey[] = "last_paid_at";
+constexpr char kSkuOrderStatusKey[] = "status";
+
+int GetVersionComponent(size_t index) {
```

Comment:
Tests could maybe call this util rather components()[index], your thoughts?

Reply:
Fixed